File: BukkitUtil-core/src/main/java/no/kh498/util/jackson/serializers/bean/ConfigurationSectionSerializer.java
package no.kh498.util.jackson.serializers.bean;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.SerializerProvider;
import com.fasterxml.jackson.databind.jsontype.TypeSerializer;
import com.fasterxml.jackson.databind.ser.std.StdSerializer;
import org.bukkit.configuration.ConfigurationSection;

import java.io.IOException;

/**
 * @author Elg
 */
public class ConfigurationSectionSerializer extends StdSerializer<ConfigurationSection> {


    static { inst = new ConfigurationSectionSerializer(); }

    public static final ConfigurationSectionSerializer inst;

    protected ConfigurationSectionSerializer() {
        super(ConfigurationSection.class);
    }

    @Override
    public void serialize(ConfigurationSection value, JsonGenerator gen, SerializerProvider provider)
    throws IOException {
        gen.writeObject(value.getValues(false));
    }

    @Override
    public void serializeWithType(ConfigurationSection value, JsonGenerator gen, SerializerProvider serializers,
                                  TypeSerializer typeSer) throws IOException {
        gen.writeObject(value.getValues(false));
    }
}

File: BukkitUtil-core/src/main/kotlin/no/kh498/util/jackson/serializers/bean/ConfigurationSectionSerializer.kt
package no.kh498.util.jackson.serializers.bean

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.SerializerProvider
import com.fasterxml.jackson.databind.jsontype.TypeSerializer
import com.fasterxml.jackson.databind.ser.std.StdSerializer
import org.bukkit.configuration.ConfigurationSection
import java.io.IOException

/**
 * @author Elg
 */
object ConfigurationSectionSerializer : StdSerializer<ConfigurationSection>(ConfigurationSection::class.java) {

    @Throws(IOException::class)
    override fun serialize(value: ConfigurationSection, gen: JsonGenerator, provider: SerializerProvider) {
        gen.writeObject(value.getValues(false))
    }

    @Throws(IOException::class)
    override fun serializeWithType(value: ConfigurationSection, gen: JsonGenerator, serializers: SerializerProvider,
                                   typeSer: TypeSerializer) {
        gen.writeObject(value.getValues(false))
    }
}
